dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java

```diff
@@ -208,7 +189,15 @@ private void registerStatedUrl(URL registryUrl, URL registeredProviderUrl, boole
 
         // url to registry
         final Registry registry = getRegistry(registryUrl);
-        final URL registeredProviderUrl = getUrlToRegistry(providerUrl, registryUrl);
+
+        URL regUrl = getUrlToRegistry(providerUrl, registryUrl);
+
+        String dockerHostBind  = ConfigUtils.getSystemProperty(DOCKER_DUBBO_IP_TO_BIND);
```

Comment:
I think The `DUBBO_IP_TO_BIND` and `DUBBO_IP_TO_REGISTRY` jvm properties have already covered this scenario.